jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala

```diff
@@ -296,7 +296,13 @@ class XGBoostClassificationModel private[ml](
           if (batchCnt == 0) {
             val rabitEnv = Array(
               "DMLC_TASK_ID" -> TaskContext.getPartitionId().toString,
-              "DMLC_WORKER_STOP_PROCESS_ON_ERROR" -> "false").toMap
+              "DMLC_WORKER_CONNECT_RETRY" -> String.valueOf(getConnectRetry),
+              "DMLC_WORKER_STOP_PROCESS_ON_ERROR" -> "false",
+              "rabit_bootstrap_cache" -> String.valueOf(getBootstrapCache),
+              "rabit_debug" -> String.valueOf(getRabitDebug),
+              "rabit_reduce_buffer" -> String.valueOf(getReduceBuffer),
+              "rabit_reduce_ring_mincount" -> String.valueOf(getRingReduceMin)
```

Comment:
let's build a common class for both classifier and regressor and put these values there

Reply:
sounds good, will do.

Reply:
done